openraft/src/core/admin.rs

```diff
@@ -179,7 +175,8 @@ impl<'a, C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> LeaderS
         turn_to_learner: bool,
         tx: RaftRespTx<ClientWriteResponse<C>, ClientWriteError<C::NodeId>>,
     ) -> Result<(), StorageError<C::NodeId>> {
-        let members = change_members.apply_to(self.core.effective_membership.membership.get_configs().last().unwrap());
+        let members = change_members
+            .apply_to(self.core.engine.state.effective_membership.membership.get_configs().last().unwrap());
```

Comment:
too long to read....

Reply:
Hmm... most of the algorithm level logic will be rewritten in the next several PR, by moving them into Engine.